### DISCUSSION TITLE
Release date 3.x

### DISCUSSION THREAD

**Discussion by Leroy1112** [lexik/LexikJWTAuthenticationBundle#1079] [General]
Hi I was wondering when we can expect the release of a 3.x branch

Im using you're bundle in my project. I want to upgrade Symfony to 6.x. But I need the 3.x release because the 2.x still uses AuthenticationProviderInterface from symfony.  I saw it is removed in 3.x.

**Comment by darthf1** [comment]
SF 6 is supported in V2. 

https://github.com/lexik/LexikJWTAuthenticationBundle/blob/2.x/Resources/doc/index.rst#symfony-53-and-higher

https://github.com/lexik/LexikJWTAuthenticationBundle/blob/2.x/composer.json#L45

**Comment by Leroy1112** [comment]
Yes and no. Yes If we believe composer. But when I update to 6.x symfony/security-core also updates to 6.x. In 6.0 they remove AuthenticationProviderInterface. Which is still being used in V2.

**Reply by darthf1** [reply]
I'm on 6.1 with V2 of this bundle so it's not just theoretical.

Did you check my link regarding configuration?

**Reply by Leroy1112** [reply]
Yes I did. And its the same configuration. 
Im using a self written container validation tool which detects if 3th party packages does not has any vulnerabilities or classes which does not exist. 
This tool is showing me that AuthenticationProviderInterface is removed in 6.0. and when im using v2 with symfony 6. it will fail.

I've seen that the interface is removed in v3. Thats why I asked when it is released. Because this will fix it for me

But when is 3.0 planned to be released ?

**Reply by chalasr** [reply]
There is no fixed release date for 3.0. As said by @darthf1, this bundle works without `AuthenticationProviderInterface` as that interface is used only on the relevant Symfony versions (where it is not deprecated).

**Reply by Leroy1112** [reply]
Classes which implements AuthenticationProviderInterface should not be loaded if the interface is removed. I would expect something like this in the extenstion.  Same goes for AuthenticatorInterface

This will fix the problem 

```
        if (!class_exists(AuthenticationProviderInterface::class)) {
            $container->removeDefinition('lexik_jwt_authentication.security.authentication.provider');
        }

        if (!class_exists(AuthenticatorInterface::class)) {
            $container->removeDefinition('lexik_jwt_authentication.security.guard.jwt_token_authenticator');
        }
```
